plugins/inventory/kubevirt.py

```diff
@@ -611,113 +621,109 @@ def populate_inventory_from_namespace(
         for name, vmi in vmis.items():
             if name in vms:
                 continue
-            hostname = self.add_host(vmi, opts.host_format, namespace_group)
+            hostname = self.add_host(vmi["metadata"], opts.host_format, namespace_group)
             self.set_vars_from_vmi(hostname, vmi, services, opts)
             self.set_composable_vars(hostname)
 
-    def add_host(
-        self, obj: ResourceField, host_format: str, namespace_group: str
-    ) -> str:
+    def add_host(self, metadata: Dict, host_format: str, namespace_group: str) -> str:
         """
         add_hosts adds a host to the inventory.
         """
         hostname = host_format.format(
-            namespace=obj.metadata.namespace,
-            name=obj.metadata.name,
-            uid=obj.metadata.uid,
+            namespace=metadata["namespace"],
```

Comment:
I would unify the way how you access the variables, because once you are using `obj["attribute"]`, then you are using `obj.get("attribute")`

Reply:
I tend to use `get` in cases where a `None` return value is acceptable. Accessing objects with brackets will raise a `KeyError` if the key is not present. In this regard Python is unlike golang, in which you always get a default value returned when accessing a map with brackets.